### DISCUSSION TITLE
Switch a connection in Transient

### DISCUSSION THREAD

**Discussion by rede676** [SpiceSharp/SpiceSharp#217] [General]
I am looking to switch the connection during a transient simulation, to make something like a SPDT switch.
I am using a voltage source as the switch body:
`voltageSoruce = new VoltageSource(voltageSoruceName, INPUT_NAME, THROW1_NAME, 0);`
and then using `voltageSoruce.Connect(INPUT_NAME, THROW2_NAME);` to switch between the 2 terminals and this works great if I run a DC simulation -> switch() -> run a DC simulation

Is there a way I can do this inside of a transient simulation while it's running?

**Comment by svenboulanger** [comment]
I think your safest bet is to use the `VoltageSwitch` component. With two switches you should be able to make an SPDT switch.

Adding/removing components in the middle of a transient simulation is probably a tough thing to do. You can maybe emulate it by restarting the simulation with the right initial conditions, but I think you'll be up for a lot of work then. The problem is that the solver is already set up and optimized while running the simulation and changing the matrix solver mid-way is currently not possible.
